streams/src/test/java/org/apache/kafka/streams/integration/StoreQueryIntegrationTest.java

```diff
@@ -180,7 +183,7 @@ public void shouldQuerySpecificActivePartitionStores() throws Exception {
 
         // Assert that all messages in the first batch were processed in a timely manner
         assertThat(semaphore.tryAcquire(batch1NumMessages, 60, TimeUnit.SECONDS), is(equalTo(true)));
-        until(() -> {
+        retryUntil(() -> {
```

Comment:
In the PR description you state:

> The first reason is that the test uses the until method which just fails whenever a retryable error occurs.

I doubt this statement, because within the `until()` the code verifies for retrieable exceptions and returns `false` if it encounters a retrieable exception which would trigger a further iteration over the code passed into the `until()`.
You can verify this behavior by running the following test:
```
public void testUntil() throws Exception {
  until(() -> {
      try {
          throw new InvalidStateStorePartitionException(
              "The specified partition 1 for store source-table does not exist.");
      } catch (final InvalidStateStoreException exception) {
          verifyRetriableException(exception);
          LOG.info("Either streams wasn't running or a re-balancing took place. Will try again.");
          return false;
      }
  });
}
``` 
This code will retry throwing the exception until the timeout is exceeded.
You will  notice that when the timeout is exceeded, the test passes. This is definitely a bug!
The following rewrite to `until()` will fix this bug:
```
  private static void until(final TestCondition condition) {
      boolean success = false;
      final long deadline = System.currentTimeMillis() + IntegrationTestUtils.DEFAULT_TIMEOUT;
      boolean deadlineExceeded = System.currentTimeMillis() >= deadline;
      while (!success && !deadlineExceeded) {
          try {
              success = condition.conditionMet();
              Thread.sleep(500L);
          } catch (final RuntimeException e) {
              throw e;
          } catch (final Exception e) {
              throw new RuntimeException(e);
          } finally {
              deadlineExceeded = System.currentTimeMillis() >= deadline;
          }
      }
      if (deadlineExceeded) {
          fail("Test execution timed out");
      }
  }
``` 

In summary, I think method `until()` does not cause any flakiness.

If you agree with my argumentation could you remove again `retryUntil()`, add the fix to `until()`, and adapt the PR description?